woocommerce/woocommerce
# Fixed `watch:build` Race Conditions & Added Convenience Watch Commands

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

There is currently a bug in `wireit` that prevents us from using it to `--watch` a task when using tool-specific commands like `webpack --watch` on other projects. While we wait for an upstream fix to land, this switches it out with `nodemon` for watching the files that change as a result of builds.

I've also added two `watch:build` commands for convenience to `@woocommerce/plugin-woocommerce`. There is `watch:build:blocks` and `watch:build:admin` that limit what they watch to what is related. 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build`
2. After the build finishes, make arbitrary changes and make sure that no matter what the watch doesn't stop.
3. Run `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build:blocks` and confirm it limits the scope of watching.
4. Run `pnpm --filter='@woocommerce/plugin-woocommerce' watch:build:admin` and confirm it limits the scope of watching.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

Only a change to development tooling.


</details>
